src/api/app/controllers/statistics/maintenance_statistics_controller.rb

```diff
@@ -3,7 +3,7 @@ class MaintenanceStatisticsController < ApplicationController
     skip_before_action :extract_user
 
     def index
-      @project = Project.find_by(name: params[:project])
+      @project = Project.find_by!(name: params[:project])
```

Comment:
authentication != authorization :-) This does not need authentication (should be public) but it needs authorization (should not show hidden project).  I would suggest you use Project.find_by_name.

**NOTE** this is not a built in magic finder but something self implemented.